src/test/operators/join_test_runner.cpp

```diff
@@ -207,19 +219,34 @@ class JoinTestRunner : public BaseTestWithParam<JoinTestConfiguration> {
           return std::vector<JoinTestConfiguration>{};
         }
 
+        // For index join executions, scenarios shall be tested in which the majority of segments are indexed but some
+        // chunks are not indexed.
+        std::vector<ChunkPosList> missing_index_lists{
+            // no missing indexes
+            ChunkPosList{},
+            // an index for the first segment is missing
+            ChunkPosList{ChunkPosition::First},
+            // an index for a segment that is not the first or last one is missing
+            ChunkPosList{ChunkPosition::Inner},
+            // an index for the last segment is missing
+            ChunkPosList{ChunkPosition::Last}};
```

Comment:
Having four different options here increases the number of test cases a lot. We have almost 4000 join index test cases now. To address https://github.com/hyrise/hyrise/pull/1851#discussion_r409575868, we also need an additional configuration parameter for "single chunk reference guarantee". This is important when it comes to calculating the number of used indexes: If the indexed table is a reference table, we currently only use an index for a segment if the join mode is `Inner` and the reference segment's PosList references a single chunk. Currently, the join test runner does not create scenarios with indexed reference tables that have segments with this "single chunk reference guarantee".

Reply:
Is there anything in the code where the handling for the first or last index missing is different? If not, I'd say that given the chunk-based nature of the approach, a single configuration with `0% < x < 100%` chunks not being indexed should be sufficient.

Regarding the single chunk PosList - can't we make the reference table contain both a single and a multi chunk PosList?

Reply:
Now two different shares of indexed segments are provided: 100% and 10%. The indexed segments are at the beginning of the column. cdf931d3370b0c4855cb5173e6616537d6ac18d0